st2api/st2api/controllers/v1/rules.py

```diff
@@ -90,6 +91,9 @@ def post(self, rule):
                                                                        rule_api=rule)
 
             rule_db = Rule.add_or_update(rule_db)
+            # After the rule has been added modify the ref_count. This way a failure to add
+            # the rule due to violated constraints will have no impact on ref_count.
+            increment_trigger_ref_count(rule_api=rule)
```

Comment:
Not related or a blocker for this PR (also not your fault) and something which has already bothered me in the past - wish we had a single code path / abstractions which was executed in the API and during the resource registrar run...

Needing to copy and paste it across multiple files and places is a disaster waiting to happen...